src/components/canvas-tools/geometry-tool.tsx

```diff
@@ -366,6 +399,21 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
       else {
         this.dragPts[id] = { initial: coords };
         this.lastPointDown = { evt, coords };
+
+        // click on selected element - deselect unless appropriate modifier key is down
```

Comment:
Looks like this does the opposite of the comment - it *only* deselects if the modifier key is down